### PR TITLE
enh(dashboads): enhancements for the open ticket widgets

### PR DESCRIPTION
**Pull request by Med1233**
## Description

- enhance “opened on” date formatting

- when opening a ticket, modal should be bigger

- replace “open ticket” wording by “ticket”

- case wording fix: standardize “Hide services…” and “hide services…”

- widget props: collapse open tickets props

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Thebarda — APPROVED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -19164,7 +19164,7 @@ msgstr "Sélectionner les sévérités des services"
 msgid "Select host severities"
 msgstr "Sélectionner les sévérités des hôtes"
 
-msgid "Enable Open Tickets"
+msgid "Enable tickets"
 msgstr "Activer Open Tickets"
```
Comment (Med1233) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19168]:
```suggestion
msgstr "Activer les tickets"
```

File: centreon/www/widgets/src/centreon-widget-resourcestable/src/Listing/translatedLabels.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ export const labelForcedCheckDescription =
 export const labelCheckDescription =
   'Check selected resources only within configured check period';
 export const labelCheck = 'Check';
-export const labelOpenTicket = 'Open ticket';
+export const labelTicket = 'Ticket';
```
Comment (Med1233) [inline centreon/www/widgets/src/centreon-widget-resourcestable/src/Listing/translatedLabels.ts:76]:
```suggestion
export const labelTicket = 'Tickets';
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Autocomplete/ConnectedAutocomplete.tsx [hunk 3]
```diff
@@ -53,6 +54,7 @@ const ConnectedAutocomplete = ({
           limitTags={2}
           value={value}
           onChange={changeValue}
+          required= {isRequiredProperty}
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Autocomplete/ConnectedAutocomplete.tsx:57]:
```suggestion
          required={isRequiredProperty}
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Autocomplete/ConnectedAutocomplete.tsx [hunk 4]
```diff
@@ -68,6 +70,7 @@ const ConnectedAutocomplete = ({
           placeholder=""
           value={value}
           onChange={changeValues}
+          required= {isRequiredProperty}
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Autocomplete/ConnectedAutocomplete.tsx:73]:
```suggestion
          required={isRequiredProperty}
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Warning.cypress.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  });
+};
+
+const label = "Warning message!" ; 
+
+describe('WidgetTextField', () => {
+  it('displays the text field', () => {
+    initialize({label });
+
+    cy.contains(label);
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Warning.cypress.spec.tsx:45]:
```suggestion
    cy.contains(label).should('be.visible');
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Warning.cypress.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      </Provider>
+    )
+  });
+};
+
+const label = "Warning message!" ; 
+
+describe('WidgetTextField', () => {
+  it('displays the text field', () => {
+    initialize({label });
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Warning.cypress.spec.tsx:43]:
```suggestion
    initialize({ label });
```

File: centreon/www/widgets/src/centreon-widget-resourcestable/src/spec/ResourcesTable.cypress.spec.tsx [hunk 7]
```diff
@@ -495,7 +494,7 @@ describe('Open tickets', () => {
 
     cy.waitForRequest('@getResources');
 
-    cy.contains('Open ticket');
+    cy.contains('Ticket');
```
Comment (Thebarda) [inline centreon/www/widgets/src/centreon-widget-resourcestable/src/spec/ResourcesTable.cypress.spec.tsx:497]:
```suggestion
    cy.contains('Ticket').should('be.visible');
```

**Review by tuntoja — APPROVED** [review]
approved feature.json

**Review by cg-tw — APPROVED** [review]

File: centreon-open-tickets/www/modules/centreon-open-tickets/views/rules/closeTicket/action.php [hunk 8]
```diff
@@ -58,7 +58,10 @@ function isSelectionValid(string $selection): bool
 $db = $dependencyInjector['configuration_db'];
 $rule = new Centreon_OpenTickets_Rule($db);
 
-$data = isset($_POST['data']) ? json_decode($_POST['data'], true) : null;
+$request = \Symfony\Component\HttpFoundation\Request::createFromGlobals();
+$payload = json_decode($request->getContent(), true);
+
+$data = isset($payload['data']) ? $payload['data'] : null;
```
Comment (jeremyjaouen) [inline centreon-open-tickets/www/modules/centreon-open-tickets/views/rules/closeTicket/action.php:64]:
```suggestion
$data = $payload['data'] ?? null;
```

**Review by Thebarda — APPROVED** [review]

**Review by cg-tw — DISMISSED** [review]

**Review by jeremyjaouen — APPROVED** [review]
LGTM For PHP

File: centreon-open-tickets/www/modules/centreon-open-tickets/views/rules/closeTicket/action.php [hunk 9]
```diff
@@ -70,7 +73,7 @@ function isSelectionValid(string $selection): bool
 
 if (
     ! isset($data['rule_id'])
-    || ! is_int($data['rule_id'])
+    || ! is_int((int) $data['rule_id'])
```
Comment (Drago37) [inline centreon-open-tickets/www/modules/centreon-open-tickets/views/rules/closeTicket/action.php:76]: Hummm strange, use is_int() and the cas in same time is very strange :) you will have a integer all the time ;) You can remove the cast ;)

**Review by Drago37 — CHANGES_REQUESTED** [review]

File: centreon-open-tickets/www/modules/centreon-open-tickets/views/rules/closeTicket/action.php (hunk 9 above)
Comment (Drago37) [reply centreon-open-tickets/www/modules/centreon-open-tickets/views/rules/closeTicket/action.php]: ok

**Review by Drago37 — APPROVED** [review]

**Review by MedWassCh — APPROVED** [review]

**Review by quborg — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]
approved features.json changes

**Pull request merged**